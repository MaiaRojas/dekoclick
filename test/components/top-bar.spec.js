'use strict';


import React from 'react';
import { render, shallow } from 'enzyme';
import sinon from 'sinon';
import TopBar from '../../src/components/top-bar';


describe('<TopBar />', () => {

  it('should print warning in console.error when missing title', () => {
    const stub = sinon.stub(console, 'error');
    const component = render(<TopBar />);
		expect(stub.calledOnce).toBe(true);
    expect(stub.getCall(0).args[0]).toMatchSnapshot();
    console.error.restore();
	});

  it('should render to a <header> containing an <h2> with the title', () => {
    const component = render(<TopBar title="Blah blah blah" />);
    const $header = component.children().first();
    expect($header.get(0).tagName).toBe('header');
    expect($header.hasClass('mui-fixed')).toBe(true);
    expect(component.find('h2').text()).toBe('Blah blah blah');
  });

  it('should have classes prop injected and title prop on instance', () => {
    const component = shallow(<TopBar title="foo" />);
    expect(component).toHaveLength(1);
    expect(component.props()).toMatchSnapshot()
    expect(component.instance().props).toEqual({ title: 'foo' });
  });

});
