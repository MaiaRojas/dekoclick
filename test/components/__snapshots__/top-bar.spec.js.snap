// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should have classes prop injected and title prop on instance 1`] = `
Object {
  "classes": Object {
    "appBar": "TopBar-appBar-1",
  },
  "title": "foo",
}
`;

exports[`<TopBar /> should print warning in console.error when missing title 1`] = `
"Warning: Failed prop type: The prop \`title\` is marked as required in \`TopBar\`, but its value is \`undefined\`.
    in TopBar (created by withStyles(TopBar))
    in withStyles(TopBar)"
`;
